types.go

```diff
@@ -1727,9 +1727,29 @@ func NewMeteredFunctionType(
 func (*FunctionType) isType() {}
 
 func (t *FunctionType) ID() string {
+	if t.typeID == "" {
+		t.typeID = t.id()
+	}
 	return t.typeID
 }
 
+func (t *FunctionType) id() string {
+	returnTypeID := t.ReturnType.ID()
+
+	switch len(t.Parameters) {
+	case 0:
+		return "(():" + returnTypeID + ")"
+	case 1:
+		return "((" + t.Parameters[0].Type.ID() + "):" + returnTypeID + ")"
+	default:
+		params := make([]string, len(t.Parameters))
+		for i, param := range t.Parameters {
+			params[i] = param.Type.ID()
+		}
+		return "((" + strings.Join(params, ", ") + "):" + returnTypeID + ")"
+	}
+}
```

Comment:
I think it is better to leave this out. We should rather consider removing `FunctionType.typeID` (in a separate issue/PR)

Reply:
@turbolent 

CCF codec doesn't encode type ID for `FunctionType` anymore, so the decoded Cadence `FunctionType` wouldn't have type ID if it can't be generated.

If we leave `id()` out, would that cause any problems when we decode CCF encoded data into Cadence values/types, and then encode those values/types in JSON-CDC?